src/MongoDB/Manager.c

```diff
@@ -893,7 +893,11 @@ static HashTable* php_phongo_manager_get_debug_info(zval* object, int* is_temp T
 	for (i = 0; i < n; i++) {
 		zval obj;
 
-		php_phongo_server_to_zval(&obj, sds[i]);
+		if (!php_phongo_server_to_zval(&obj, sds[i])) {
+			/* Exception already thrown */
+			goto failure;
```

Comment:
Outstanding leaks are due to fields on `php_phongo_server_to_zval`'s retval not being freed. I think we can sort that out with `zval_ptr_dtor(&obj);` before the `goto` here. Likewise for the PHP 5.6 branch.